packages/venia-ui/lib/components/SavedPaymentsPage/creditCard.js

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Trash2 as DeleteIcon } from 'react-feather';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+
+import defaultClasses from './creditCard.css';
+
+/**
+ * Enumerated list of supported credit card types from
+ *
+ * https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js
+ */
+const cardTypeMapper = {
+    AE: 'American Express',
+    AU: 'Aura',
+    DI: 'Discover',
+    DN: 'Diners',
+    ELO: 'Elo',
+    HC: 'Hipercard',
+    JCB: 'JCB',
+    MC: 'MasterCard',
+    MD: 'Maestro Domestic',
+    MI: 'Maestro International',
+    UN: 'UnionPay',
+    VI: 'Visa'
+};
+
+const CreditCard = props => {
+    const { classes: propClasses, details, public_hash } = props;
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const number = `**** ${details.maskedCC} \u00A0\u00A0 ${
+        cardTypeMapper[details.type]
```

Comment:
Just in case let's not render `undefined` to the screen. Another reason we should also consider not using unicode here to get the spacing you want; consider using css.
```suggestion
        cardTypeMapper[details.type] || ''
```